### PR TITLE
Add strict values validation on json schema

### PR DESCRIPTION
**Pull request by mavimo**
### Describe your changes :

I worked on the enforce values at schema level because it reduce the possibility of misconfiguration from the final users

#
### Type of change :

- [x] Improvement
- [x] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@akash-jain-10

### CODE REVIEW

**Comment by akash-jain-10** [comment]
Hello @mavimo would you like to take this one ? - We have this values validations listed down as sub tasks -https://github.com/open-metadata/openmetadata-helm-charts/issues/84

**Comment by akash-jain-10** [comment]
> @akash-jain-10 sure, some are already solved in the current PR, other can be solved innsequential PR or do you suggest to have all of them solved in this single PR?

It would be great to get all of these available in one release of Helm Chart!

**Comment by mavimo** [comment]
@akash-jain-10 Added other validations, not sure is doable to add dependencies for items that are out of the same tree (in auth).
Seems possible to add dependencies between value of the field and other fields if they are in the same node of the tree, but in this case `global.airflow.openmetadata.authProvider`  need to drive the value of `global.airflow.openmetadata.authConfig.azure` that is a subitem of another node.

https://stackoverflow.com/questions/38717933/jsonschema-attribute-conditionally-required

I'll investigate a bit more and than I will try to report my investigation here.

**Comment by akash-jain-10** [comment]
> @akash-jain-10 Added other validations, not sure is doable to add dependencies for items that are out of the same tree (in auth). Seems possible to add dependencies between value of the field and other fields if they are in the same node of the tree, but in this case `global.airflow.openmetadata.authProvider` need to drive the value of `global.airflow.openmetadata.authConfig.azure` that is a subitem of another node.
> 
> https://stackoverflow.com/questions/38717933/jsonschema-attribute-conditionally-required
> 
> I'll investigate a bit more and then I will try to report my investigation here.
> 
> _EDIT_: I try to create an simplified version here: https://www.jsonschemavalidator.net/s/KNo19685 but seems not doable to validate sub-property from the top level, do you have some suggestion?

Hello @mavimo . Looks like we might need to get this available via `validate-values.tpl` file. Ref - https://github.com/open-metadata/openmetadata-helm-charts/blob/main/charts/openmetadata/templates/validate-values.tpl

Another similar example without using json-schema. https://github.com/airflow-helm/charts/blob/main/charts/airflow/templates/_helpers/validate-values.tpl

**Comment by mavimo** [comment]
@akash-jain-10 thx to pointing me out to this part of the template.. is't a bit different solution (validation is not done at values level but when is going to be applied ) but I can work on it.

It's fine to remove validation that are done already at the schema level?

**Comment by mavimo** [comment]
@akash-jain-10 should be fine but i will like to add some tests in order to check that it work as expected, can you wait to merge it? 🙏

**Comment by akash-jain-10** [comment]
> @akash-jain-10 should be fine but i will like to add some tests in order to check that it work as expected, can you wait to merge it? 🙏

Yes. That would be great! Thanks.

**Comment by mavimo** [comment]
@akash-jain-10 added some tests (add the `helm unittest` plugin for that).

I need rename `validate-values.tpl` into `validate-values.yaml`, the reason behind this change is that helm do not render non YAML file and this make impossible to test it, see more info [here](https://github.com/helm/helm/issues/7899)

There still an issue on tests (I want to test that the config are not required if airflow is not installed as subchart), but for some reason the test is failing and need more investigation.

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]
Hello @mavimo . With this [PR](https://github.com/open-metadata/OpenMetadata/pull/8896), we are going off with `global.airflow.openmetadata.authConfig.*` with 0.13.1 Release. Can you please update the changes on this PR to handle only the the schema related changes ?

File: charts/openmetadata/templates/validate-values.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+{{- if and (
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/validate-values.yaml:1]: Any specific reason to use the filename with `.yaml` then `.tpl` ?

Comment (mavimo) [reply charts/openmetadata/templates/validate-values.yaml:1]: Y, sorry, I think github collapsed all the intermediate section, basically:

https://github.com/open-metadata/openmetadata-helm-charts/pull/93#issuecomment-1336245686

**Comment by akash-jain-10** [comment]
Hey @mavimo the `validate-values.yaml` file shouldn't be required for now starting with `0.13.1` Release. Can you please remove them from this PR and it's associated code as well ? This will mark the Pull Request as ready to merge in. Thanks.

**Comment by mavimo** [comment]
@akash-jain-10 I'll do in the upcoming days, sorry, today I can't work on it. I'll keep you posted!

**Comment by akash-jain-10** [comment]
Hello @mavimo . Is this something we can target for upcoming release `0.13.2` scheduled at 24th January ?

**Comment by mavimo** [comment]
@akash-jain-10 unfortunately I can't fit it in the release planned for the 24, sorry, I'll try in the upcoming days :(

**Comment by akash-jain-10** [comment]
Closing this Pull Request as we have taken care of this in the Pull request [here](https://github.com/open-metadata/openmetadata-helm-charts/pull/122)

**Pull request closed without merging**